rust-runtime/inlineable/src/cbor_errors.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_cbor::decode::DeserializeError;
+use aws_smithy_cbor::Decoder;
+use aws_smithy_runtime_api::http::Headers;
+use aws_smithy_types::error::metadata::{Builder as ErrorMetadataBuilder, ErrorMetadata};
+
+fn sanitize_error_code(error_code: &str) -> &str {
+    // Trim a trailing URL from the error code, beginning with a `:`
+    let error_code = match error_code.find(':') {
```

Comment:
Since we're looking for trailing `:`, don't you want https://doc.rust-lang.org/std/primitive.str.html#method.rfind?

Reply:
If we used that, then [this unit test](https://github.com/smithy-lang/smithy-rs/blob/1223f61e10c82b5b8172ad9e18cd6cd558203751/rust-runtime/inlineable/src/json_errors.rs#L167-L174) would fail (assuming we use unaltered version of [json_errors::sanitize_error_code](https://github.com/smithy-lang/smithy-rs/blob/1223f61e10c82b5b8172ad9e18cd6cd558203751/rust-runtime/inlineable/src/json_errors.rs#L18) in `cbor_errors`).

Reply:
Added supplementary comment in 0e153b3.